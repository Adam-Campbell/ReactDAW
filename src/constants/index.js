export * from './UIColors';
export * from './programData';
export * from './instrumentData';
export * from './effectData';