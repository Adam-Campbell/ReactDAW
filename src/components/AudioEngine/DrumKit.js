import Tone from 'tone';
import { drumSampleURLs } from '../../constants';
import { createPitchesArray } from '../../sharedUtils';

class DrumKit {
    constructor() {
        this.outputNode = new Tone.Volume();
        this.drums = {
            kick: new Tone.Player(drumSampleURLs.kick001).connect(this.outputNode),
            snare: new Tone.Player(drumSampleURLs.snare001).connect(this.outputNode),
            clap: new Tone.Player(drumSampleURLs.clap001).connect(this.outputNode),
            closedHat: new Tone.Player(drumSampleURLs.closedHat001).connect(this.outputNode),
            openHat: new Tone.Player(drumSampleURLs.openHat001).connect(this.outputNode),
            crash: new Tone.Player(drumSampleURLs.crash001).connect(this.outputNode),
            ride: new Tone.Player(drumSampleURLs.ride001).connect(this.outputNode),
            highTom: new Tone.Player(drumSampleURLs.highTom001).connect(this.outputNode),
            midTom: new Tone.Player(drumSampleURLs.midTom001).connect(this.outputNode),
            lowTom: new Tone.Player(drumSampleURLs.lowTom001).connect(this.outputNode),
            fx: new Tone.Player(drumSampleURLs.fx001).connect(this.outputNode)
        };
        this.drums.kick.retrigger = true;
        this.drums.snare.retrigger = true;
        this.drums.clap.retrigger = true;
        this.drums.closedHat.retrigger = true;
        this.drums.openHat.retrigger = true;
        this.drums.crash.retrigger = true;
        this.drums.ride.retrigger = true;
        this.drums.highTom.retrigger = true;
        this.drums.midTom.retrigger = true;
        this.drums.lowTom.retrigger = true;
        this.drums.fx.retrigger = true;
        this.pitchesArray = createPitchesArray();
    }


    /**
     * Takes in a number between 0 and 10 (inclusive), and returns a corresponding string representing
     * one of the drum types.
     * @param {number} indexValue
     * @returns {string} drum type string 
     */
    mapIndexValueToDrumString(indexValue) {
        switch (indexValue) {
            case 0:
                return 'kick';
            case 1: 
                return 'snare';
            case 2:
                return 'clap';
            case 3:
                return 'closedHat';
            case 4:
                return 'openHat';
            case 5:
                return 'crash';
            case 6:
                return 'ride';
            case 7:
                return 'highTom';
            case 8:
                return 'midTom';
            case 9:
                return 'lowTom';
            case 10:
                return 'fx';
            default:
                return 'kick';
        }
    }

    /**
     * Takes the pitch of a note and maps it to a drum type.
     * @param {string} pitch - a note pitch
     * @returns {string} the drum type string
     */
    convertPitchToDrum(pitch) {
        const pitchIndex = this.pitchesArray.indexOf(pitch);
        const adjustedIndex = pitchIndex % 11;
        const drumString = this.mapIndexValueToDrumString(adjustedIndex);
        return drumString; 
    }

    /**
     * Mimics the triggerAttackRelease methods on the various instrument classes in the Tone library. 
     * Enables other parts of the program to treat this class the same as any of the Tone instrument 
     * classes. Note that duration and velocity are not required by this class, they are merely passed
     * in so that other parts of the program can call this triggerAttackRelease method the same as they
     * would on any other instrument class. 
     * @param {string} pitch - 'C5' 
     * @param {string} duration - '0:1:0' 
     * @param {string} time - '0:1:0' 
     * @param {number} velocity - 1 
     */
    triggerAttackRelease(pitch, duration, time, velocity) {
        const drum = this.convertPitchToDrum(pitch);
        this.drums[drum].start(time);
    }

    /**
     * Updates the sample loaded into one of the drums. 
     * @param {string} drum - the drum to update
     * @param {string} newSampleURL - the new sample to update it with
     */
    updateSample(drum, newSampleURL) {
        this.drums[drum].load(newSampleURL);
    }

    /**
     * Analagous to the connect method on any of the Tone instrument classes.
     * @param {object} node - a valid node that can accept connections 
     */
    connect(node) {
        // for (let key in this.drums) {
        //     this.drums[key].connect(node);
        // }
        this.outputNode.connect(node);
    }

    /**
     * Analagous to the disconnect method on any of the Tone instrument classes.
     * @param {object} node - a valid node that can accept connections 
     */
    disconnect(node) {
        // for (let key in this.drums) {
        //     this.drums[key].disconnect(node);
        // }
        this.outputNode.disconnect(node);
    }

    /**
     * Analagous to the set method on any of the Tone instrument classes.
     * @param {object} data - the new data to update the DrumKit with.
     */
    set(data) {
        for (let key in data) {
            this.updateSample(key, data[key]);
        }
    }
}

export default DrumKit;