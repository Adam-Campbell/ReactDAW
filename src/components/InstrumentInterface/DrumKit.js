import React from 'react';
import PropTypes from 'prop-types';
import HeaderModule from './HeaderModule';
import { drumSampleURLs } from '../../constants';
import DrumModule from './DrumModule';

const DrumKit = props => (
    <div className="synth-interface__container">
        <HeaderModule synthTitle="Drum Kit" />
        <div className="synth-interface__module-container">
            <DrumModule 
                drumType="kick"
                drumName="Kick"
                drumData={props.instrumentData.kick}
                instrumentId={props.instrumentId}
                handleChange={props.handleChange}
                drumSampleOptions={[
                    { value: drumSampleURLs.kick001, text: 'Kick 001' },
                    { value: drumSampleURLs.kick002, text: 'Kick 002' },
                    { value: drumSampleURLs.kick003, text: 'Kick 003' },
                    { value: drumSampleURLs.kick004, text: 'Kick 004' },
                    { value: drumSampleURLs.kick005, text: 'Kick 005' },
                    { value: drumSampleURLs.kick006, text: 'Kick 006' },
                    { value: drumSampleURLs.kick007, text: 'Kick 007' },
                    { value: drumSampleURLs.kick008, text: 'Kick 008' },
                    { value: drumSampleURLs.kick009, text: 'Kick 009' },
                    { value: drumSampleURLs.kick010, text: 'Kick 010' },
                    { value: drumSampleURLs.kick011, text: 'Kick 011' },
                    { value: drumSampleURLs.kick012, text: 'Kick 012' },
                    { value: drumSampleURLs.kick013, text: 'Kick 013' },
                    { value: drumSampleURLs.kick014, text: 'Kick 014' },
                    { value: drumSampleURLs.kick015, text: 'Kick 015' },
                    { value: drumSampleURLs.kick016, text: 'Kick 016' },
                    { value: drumSampleURLs.kick017, text: 'Kick 017' },
                    { value: drumSampleURLs.kick018, text: 'Kick 018' },
                    { value: drumSampleURLs.kick019, text: 'Kick 019' },
                    { value: drumSampleURLs.kick020, text: 'Kick 020' },
                ]}
            />
            <DrumModule 
                drumType="snare"
                drumName="Snare"
                drumData={props.instrumentData.snare}
                instrumentId={props.instrumentId}
                handleChange={props.handleChange}
                drumSampleOptions={[
                    { value: drumSampleURLs.snare001, text: 'Snare 001' },
                    { value: drumSampleURLs.snare002, text: 'Snare 002' },
                    { value: drumSampleURLs.snare003, text: 'Snare 003' },
                    { value: drumSampleURLs.snare004, text: 'Snare 004' },
                    { value: drumSampleURLs.snare005, text: 'Snare 005' },
                    { value: drumSampleURLs.snare006, text: 'Snare 006' },
                    { value: drumSampleURLs.snare007, text: 'Snare 007' },
                    { value: drumSampleURLs.snare008, text: 'Snare 008' },
                    { value: drumSampleURLs.snare009, text: 'Snare 009' },
                    { value: drumSampleURLs.snare010, text: 'Snare 010' },
                    { value: drumSampleURLs.snare011, text: 'Snare 011' },
                    { value: drumSampleURLs.snare012, text: 'Snare 012' },
                    { value: drumSampleURLs.snare013, text: 'Snare 013' },
                    { value: drumSampleURLs.snare014, text: 'Snare 014' },
                    { value: drumSampleURLs.snare015, text: 'Snare 015' },
                    { value: drumSampleURLs.snare016, text: 'Snare 016' },
                    { value: drumSampleURLs.snare017, text: 'Snare 017' },
                    { value: drumSampleURLs.snare018, text: 'Snare 018' },
                    { value: drumSampleURLs.snare019, text: 'Snare 019' },
                    { value: drumSampleURLs.snare020, text: 'Snare 020' },
                    { value: drumSampleURLs.snare021, text: 'Snare 021' },
                    { value: drumSampleURLs.snare022, text: 'Snare 022' },
                ]}   
            />
            <DrumModule 
                drumType="clap"
                drumName="Clap"
                drumData={props.instrumentData.clap}
                instrumentId={props.instrumentId}
                handleChange={props.handleChange}
                drumSampleOptions={[
                    { value: drumSampleURLs.clap001, text: 'Clap 001' },
                    { value: drumSampleURLs.clap002, text: 'Clap 002' },
                    { value: drumSampleURLs.clap003, text: 'Clap 003' },
                    { value: drumSampleURLs.clap004, text: 'Clap 004' },
                    { value: drumSampleURLs.clap005, text: 'Clap 005' },
                    { value: drumSampleURLs.clap006, text: 'Clap 006' },
                    { value: drumSampleURLs.clap007, text: 'Clap 007' },
                    { value: drumSampleURLs.clap008, text: 'Clap 008' },
                    { value: drumSampleURLs.clap009, text: 'Clap 009' },
                    { value: drumSampleURLs.clap010, text: 'Clap 010' }
                ]}
            />
        </div>
        <div className="synth-interface__module-container">
            <DrumModule 
                drumType="closedHat"
                drumName="Closed Hat"
                drumData={props.instrumentData.closedHat}
                instrumentId={props.instrumentId}
                handleChange={props.handleChange}
                drumSampleOptions={[
                    { value: drumSampleURLs.closedHat001, text: 'Closed Hat 001' },
                    { value: drumSampleURLs.closedHat002, text: 'Closed Hat 002' },
                    { value: drumSampleURLs.closedHat003, text: 'Closed Hat 003' },
                    { value: drumSampleURLs.closedHat004, text: 'Closed Hat 004' },
                    { value: drumSampleURLs.closedHat005, text: 'Closed Hat 005' },
                    { value: drumSampleURLs.closedHat006, text: 'Closed Hat 006' },
                    { value: drumSampleURLs.closedHat007, text: 'Closed Hat 007' },
                    { value: drumSampleURLs.closedHat008, text: 'Closed Hat 008' },
                    { value: drumSampleURLs.closedHat009, text: 'Closed Hat 009' },
                    { value: drumSampleURLs.closedHat010, text: 'Closed Hat 010' },
                    { value: drumSampleURLs.closedHat011, text: 'Closed Hat 011' },
                    { value: drumSampleURLs.closedHat012, text: 'Closed Hat 012' },
                    { value: drumSampleURLs.closedHat013, text: 'Closed Hat 013' },
                    { value: drumSampleURLs.closedHat014, text: 'Closed Hat 014' },
                    { value: drumSampleURLs.closedHat015, text: 'Closed Hat 015' },
                    { value: drumSampleURLs.closedHat016, text: 'Closed Hat 016' }
                ]}   
            />
            <DrumModule 
                drumType="openHat"
                drumName="Open Hat"
                drumData={props.instrumentData.openHat}
                instrumentId={props.instrumentId}
                handleChange={props.handleChange}
                drumSampleOptions={[
                    { value: drumSampleURLs.openHat001, text: 'Open Hat 001' },
                    { value: drumSampleURLs.openHat002, text: 'Open Hat 002' },
                    { value: drumSampleURLs.openHat003, text: 'Open Hat 003' },
                    { value: drumSampleURLs.openHat004, text: 'Open Hat 004' },
                    { value: drumSampleURLs.openHat005, text: 'Open Hat 005' },
                    { value: drumSampleURLs.openHat006, text: 'Open Hat 006' },
                    { value: drumSampleURLs.openHat007, text: 'Open Hat 007' },
                    { value: drumSampleURLs.openHat008, text: 'Open Hat 008' },
                    { value: drumSampleURLs.openHat009, text: 'Open Hat 009' }
                ]}   
            />
            <DrumModule 
                drumType="crash"
                drumName="Crash"
                drumData={props.instrumentData.crash}
                instrumentId={props.instrumentId}
                handleChange={props.handleChange}
                drumSampleOptions={[
                    { value: drumSampleURLs.crash001, text: 'Crash 001' },
                    { value: drumSampleURLs.crash002, text: 'Crash 002' },
                    { value: drumSampleURLs.crash003, text: 'Crash 003' },
                    { value: drumSampleURLs.crash004, text: 'Crash 004' },
                    { value: drumSampleURLs.crash005, text: 'Crash 005' }
                ]}   
            />
        </div>
        <div className="synth-interface__module-container">
            <DrumModule 
                drumType="ride"
                drumName="Ride"
                drumData={props.instrumentData.ride}
                instrumentId={props.instrumentId}
                handleChange={props.handleChange}
                drumSampleOptions={[
                    { value: drumSampleURLs.ride001, text: 'Ride 001' },
                    { value: drumSampleURLs.ride002, text: 'Ride 002' },
                    { value: drumSampleURLs.ride003, text: 'Ride 003' },
                    { value: drumSampleURLs.ride004, text: 'Ride 004' },
                    { value: drumSampleURLs.ride005, text: 'Ride 005' },
                    { value: drumSampleURLs.ride006, text: 'Ride 006' },
                    { value: drumSampleURLs.ride007, text: 'Ride 007' }
                ]}   
            />
            <DrumModule 
                drumType="highTom"
                drumName="High Tom"
                drumData={props.instrumentData.highTom}
                instrumentId={props.instrumentId}
                handleChange={props.handleChange}
                drumSampleOptions={[
                    { value: drumSampleURLs.highTom001, text: 'High Tom 001' },
                    { value: drumSampleURLs.highTom002, text: 'High Tom 002' },
                    { value: drumSampleURLs.highTom003, text: 'High Tom 003' },
                    { value: drumSampleURLs.highTom004, text: 'High Tom 004' },
                    { value: drumSampleURLs.highTom005, text: 'High Tom 005' }
                ]}   
            />
            <DrumModule 
                drumType="midTom"
                drumName="Mid Tom"
                drumData={props.instrumentData.midTom}
                instrumentId={props.instrumentId}
                handleChange={props.handleChange}
                drumSampleOptions={[
                    { value: drumSampleURLs.midTom001, text: 'Mid Tom 001' },
                    { value: drumSampleURLs.midTom002, text: 'Mid Tom 002' },
                    { value: drumSampleURLs.midTom003, text: 'Mid Tom 003' },
                    { value: drumSampleURLs.midTom004, text: 'Mid Tom 004' },
                    { value: drumSampleURLs.midTom005, text: 'Mid Tom 005' },
                    { value: drumSampleURLs.midTom006, text: 'Mid Tom 006' }
                ]}   
            />
        </div>
        <div className="synth-interface__module-container">
            <DrumModule 
                drumType="lowTom"
                drumName="Low Tom"
                drumData={props.instrumentData.lowTom}
                instrumentId={props.instrumentId}
                handleChange={props.handleChange}
                drumSampleOptions={[
                    { value: drumSampleURLs.lowTom001, text: 'Low Tom 001' },
                    { value: drumSampleURLs.lowTom002, text: 'Low Tom 002' },
                    { value: drumSampleURLs.lowTom003, text: 'Low Tom 003' },
                    { value: drumSampleURLs.lowTom004, text: 'Low Tom 004' },
                    { value: drumSampleURLs.lowTom005, text: 'Low Tom 005' }
                ]}   
            />
            <DrumModule 
                drumType="fx"
                drumName="fx"
                drumData={props.instrumentData.fx}
                instrumentId={props.instrumentId}
                handleChange={props.handleChange}
                drumSampleOptions={[
                    { value: drumSampleURLs.fx001, text: 'FX 001' },
                    { value: drumSampleURLs.fx002, text: 'FX 002' },
                    { value: drumSampleURLs.fx003, text: 'FX 003' },
                    { value: drumSampleURLs.fx004, text: 'FX 004' },
                    { value: drumSampleURLs.fx005, text: 'FX 005' },
                    { value: drumSampleURLs.fx006, text: 'FX 006' },
                    { value: drumSampleURLs.fx007, text: 'FX 007' },
                    { value: drumSampleURLs.fx008, text: 'FX 008' },
                ]}   
            />
        </div>
    </div>
);

DrumKit.propTypes = {
    instrumentData: PropTypes.object.isRequired,
    instrumentId: PropTypes.string.isRequired,
    handleChange: PropTypes.func.isRequired
};

export default DrumKit;