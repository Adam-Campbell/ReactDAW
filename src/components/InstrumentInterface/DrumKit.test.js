import React from 'react';
import { shallow } from 'enzyme';
import DrumKit from './Drumkit';
import { instrumentData } from '../../constants';

test('renders correctly', () => {
    const component = shallow(
        <DrumKit 
            instrumentData={instrumentData.drumKit}
            instrumentId="4326432946537512"
            handleChange={jest.fn()}
        />
    );
    expect(component).toMatchSnapshot();
});