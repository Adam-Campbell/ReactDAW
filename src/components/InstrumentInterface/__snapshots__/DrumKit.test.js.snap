// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="instrument-interface__container"
>
  <HeaderModule
    instrumentTitle="Drum Kit"
  />
  <div
    className="instument-interface__module-container"
  >
    <DrumModule
      drumData={
        Object {
          "pan": 0,
          "sample": "drum-samples/kicks/kick_001.wav",
          "volume": 0,
        }
      }
      drumName="Kick"
      drumSampleOptions={
        Array [
          Object {
            "text": "Kick 001",
            "value": "drum-samples/kicks/kick_001.wav",
          },
          Object {
            "text": "Kick 002",
            "value": "drum-samples/kicks/kick_002.wav",
          },
          Object {
            "text": "Kick 003",
            "value": "drum-samples/kicks/kick_003.wav",
          },
          Object {
            "text": "Kick 004",
            "value": "drum-samples/kicks/kick_004.wav",
          },
          Object {
            "text": "Kick 005",
            "value": "drum-samples/kicks/kick_005.wav",
          },
          Object {
            "text": "Kick 006",
            "value": "drum-samples/kicks/kick_006.wav",
          },
          Object {
            "text": "Kick 007",
            "value": "drum-samples/kicks/kick_007.wav",
          },
          Object {
            "text": "Kick 008",
            "value": "drum-samples/kicks/kick_008.wav",
          },
          Object {
            "text": "Kick 009",
            "value": "drum-samples/kicks/kick_009.wav",
          },
          Object {
            "text": "Kick 010",
            "value": "drum-samples/kicks/kick_010.wav",
          },
          Object {
            "text": "Kick 011",
            "value": "drum-samples/kicks/kick_011.wav",
          },
          Object {
            "text": "Kick 012",
            "value": "drum-samples/kicks/kick_012.wav",
          },
          Object {
            "text": "Kick 013",
            "value": "drum-samples/kicks/kick_013.wav",
          },
          Object {
            "text": "Kick 014",
            "value": "drum-samples/kicks/kick_014.wav",
          },
          Object {
            "text": "Kick 015",
            "value": "drum-samples/kicks/kick_015.wav",
          },
          Object {
            "text": "Kick 016",
            "value": "drum-samples/kicks/kick_016.wav",
          },
          Object {
            "text": "Kick 017",
            "value": "drum-samples/kicks/kick_017.wav",
          },
          Object {
            "text": "Kick 018",
            "value": "drum-samples/kicks/kick_018.wav",
          },
          Object {
            "text": "Kick 019",
            "value": "drum-samples/kicks/kick_019.wav",
          },
          Object {
            "text": "Kick 020",
            "value": "drum-samples/kicks/kick_020.wav",
          },
        ]
      }
      drumType="kick"
      handleChange={[MockFunction]}
      instrumentId="4326432946537512"
    />
    <DrumModule
      drumData={
        Object {
          "pan": 0,
          "sample": "drum-samples/snares/snare_001.wav",
          "volume": 0,
        }
      }
      drumName="Snare"
      drumSampleOptions={
        Array [
          Object {
            "text": "Snare 001",
            "value": "drum-samples/snares/snare_001.wav",
          },
          Object {
            "text": "Snare 002",
            "value": "drum-samples/snares/snare_002.wav",
          },
          Object {
            "text": "Snare 003",
            "value": "drum-samples/snares/snare_003.wav",
          },
          Object {
            "text": "Snare 004",
            "value": "drum-samples/snares/snare_004.wav",
          },
          Object {
            "text": "Snare 005",
            "value": "drum-samples/snares/snare_005.wav",
          },
          Object {
            "text": "Snare 006",
            "value": "drum-samples/snares/snare_006.wav",
          },
          Object {
            "text": "Snare 007",
            "value": "drum-samples/snares/snare_007.wav",
          },
          Object {
            "text": "Snare 008",
            "value": "drum-samples/snares/snare_008.wav",
          },
          Object {
            "text": "Snare 009",
            "value": "drum-samples/snares/snare_009.wav",
          },
          Object {
            "text": "Snare 010",
            "value": "drum-samples/snares/snare_010.wav",
          },
          Object {
            "text": "Snare 011",
            "value": "drum-samples/snares/snare_011.wav",
          },
          Object {
            "text": "Snare 012",
            "value": "drum-samples/snares/snare_012.wav",
          },
          Object {
            "text": "Snare 013",
            "value": "drum-samples/snares/snare_013.wav",
          },
          Object {
            "text": "Snare 014",
            "value": "drum-samples/snares/snare_014.wav",
          },
          Object {
            "text": "Snare 015",
            "value": "drum-samples/snares/snare_015.wav",
          },
          Object {
            "text": "Snare 016",
            "value": "drum-samples/snares/snare_016.wav",
          },
          Object {
            "text": "Snare 017",
            "value": "drum-samples/snares/snare_017.wav",
          },
          Object {
            "text": "Snare 018",
            "value": "drum-samples/snares/snare_018.wav",
          },
          Object {
            "text": "Snare 019",
            "value": "drum-samples/snares/snare_019.wav",
          },
          Object {
            "text": "Snare 020",
            "value": "drum-samples/snares/snare_020.wav",
          },
          Object {
            "text": "Snare 021",
            "value": "drum-samples/snares/snare_021.wav",
          },
          Object {
            "text": "Snare 022",
            "value": "drum-samples/snares/snare_022.wav",
          },
        ]
      }
      drumType="snare"
      handleChange={[MockFunction]}
      instrumentId="4326432946537512"
    />
    <DrumModule
      drumData={
        Object {
          "pan": 0,
          "sample": "drum-samples/claps/clap_001.wav",
          "volume": 0,
        }
      }
      drumName="Clap"
      drumSampleOptions={
        Array [
          Object {
            "text": "Clap 001",
            "value": "drum-samples/claps/clap_001.wav",
          },
          Object {
            "text": "Clap 002",
            "value": "drum-samples/claps/clap_002.wav",
          },
          Object {
            "text": "Clap 003",
            "value": "drum-samples/claps/clap_003.wav",
          },
          Object {
            "text": "Clap 004",
            "value": "drum-samples/claps/clap_004.wav",
          },
          Object {
            "text": "Clap 005",
            "value": "drum-samples/claps/clap_005.wav",
          },
          Object {
            "text": "Clap 006",
            "value": "drum-samples/claps/clap_006.wav",
          },
          Object {
            "text": "Clap 007",
            "value": "drum-samples/claps/clap_007.wav",
          },
          Object {
            "text": "Clap 008",
            "value": "drum-samples/claps/clap_008.wav",
          },
          Object {
            "text": "Clap 009",
            "value": "drum-samples/claps/clap_009.wav",
          },
          Object {
            "text": "Clap 010",
            "value": "drum-samples/claps/clap_010.wav",
          },
        ]
      }
      drumType="clap"
      handleChange={[MockFunction]}
      instrumentId="4326432946537512"
    />
  </div>
  <div
    className="instrument-interface__module-container"
  >
    <DrumModule
      drumData={
        Object {
          "pan": 0,
          "sample": "drum-samples/closed-hats/closed-hat_001.wav",
          "volume": 0,
        }
      }
      drumName="Closed Hat"
      drumSampleOptions={
        Array [
          Object {
            "text": "Closed Hat 001",
            "value": "drum-samples/closed-hats/closed-hat_001.wav",
          },
          Object {
            "text": "Closed Hat 002",
            "value": "drum-samples/closed-hats/closed-hat_002.wav",
          },
          Object {
            "text": "Closed Hat 003",
            "value": "drum-samples/closed-hats/closed-hat_003.wav",
          },
          Object {
            "text": "Closed Hat 004",
            "value": "drum-samples/closed-hats/closed-hat_004.wav",
          },
          Object {
            "text": "Closed Hat 005",
            "value": "drum-samples/closed-hats/closed-hat_005.wav",
          },
          Object {
            "text": "Closed Hat 006",
            "value": "drum-samples/closed-hats/closed-hat_006.wav",
          },
          Object {
            "text": "Closed Hat 007",
            "value": "drum-samples/closed-hats/closed-hat_007.wav",
          },
          Object {
            "text": "Closed Hat 008",
            "value": "drum-samples/closed-hats/closed-hat_008.wav",
          },
          Object {
            "text": "Closed Hat 009",
            "value": "drum-samples/closed-hats/closed-hat_009.wav",
          },
          Object {
            "text": "Closed Hat 010",
            "value": "drum-samples/closed-hats/closed-hat_010.wav",
          },
          Object {
            "text": "Closed Hat 011",
            "value": "drum-samples/closed-hats/closed-hat_011.wav",
          },
          Object {
            "text": "Closed Hat 012",
            "value": "drum-samples/closed-hats/closed-hat_012.wav",
          },
          Object {
            "text": "Closed Hat 013",
            "value": "drum-samples/closed-hats/closed-hat_013.wav",
          },
          Object {
            "text": "Closed Hat 014",
            "value": "drum-samples/closed-hats/closed-hat_014.wav",
          },
          Object {
            "text": "Closed Hat 015",
            "value": "drum-samples/closed-hats/closed-hat_015.wav",
          },
          Object {
            "text": "Closed Hat 016",
            "value": "drum-samples/closed-hats/closed-hat_016.wav",
          },
        ]
      }
      drumType="closedHat"
      handleChange={[MockFunction]}
      instrumentId="4326432946537512"
    />
    <DrumModule
      drumData={
        Object {
          "pan": 0,
          "sample": "drum-samples/open-hats/open-hat_001.wav",
          "volume": 0,
        }
      }
      drumName="Open Hat"
      drumSampleOptions={
        Array [
          Object {
            "text": "Open Hat 001",
            "value": "drum-samples/open-hats/open-hat_001.wav",
          },
          Object {
            "text": "Open Hat 002",
            "value": "drum-samples/open-hats/open-hat_002.wav",
          },
          Object {
            "text": "Open Hat 003",
            "value": "drum-samples/open-hats/open-hat_003.wav",
          },
          Object {
            "text": "Open Hat 004",
            "value": "drum-samples/open-hats/open-hat_004.wav",
          },
          Object {
            "text": "Open Hat 005",
            "value": "drum-samples/open-hats/open-hat_005.wav",
          },
          Object {
            "text": "Open Hat 006",
            "value": "drum-samples/open-hats/open-hat_006.wav",
          },
          Object {
            "text": "Open Hat 007",
            "value": "drum-samples/open-hats/open-hat_007.wav",
          },
          Object {
            "text": "Open Hat 008",
            "value": "drum-samples/open-hats/open-hat_008.wav",
          },
          Object {
            "text": "Open Hat 009",
            "value": "drum-samples/open-hats/open-hat_009.wav",
          },
        ]
      }
      drumType="openHat"
      handleChange={[MockFunction]}
      instrumentId="4326432946537512"
    />
    <DrumModule
      drumData={
        Object {
          "pan": 0,
          "sample": "drum-samples/crashes/crash_001.wav",
          "volume": 0,
        }
      }
      drumName="Crash"
      drumSampleOptions={
        Array [
          Object {
            "text": "Crash 001",
            "value": "drum-samples/crashes/crash_001.wav",
          },
          Object {
            "text": "Crash 002",
            "value": "drum-samples/crashes/crash_002.wav",
          },
          Object {
            "text": "Crash 003",
            "value": "drum-samples/crashes/crash_003.wav",
          },
          Object {
            "text": "Crash 004",
            "value": "drum-samples/crashes/crash_004.wav",
          },
          Object {
            "text": "Crash 005",
            "value": "drum-samples/crashes/crash_005.wav",
          },
        ]
      }
      drumType="crash"
      handleChange={[MockFunction]}
      instrumentId="4326432946537512"
    />
  </div>
  <div
    className="instrument-interface__module-container"
  >
    <DrumModule
      drumData={
        Object {
          "pan": 0,
          "sample": "drum-samples/rides/ride_001.wav",
          "volume": 0,
        }
      }
      drumName="Ride"
      drumSampleOptions={
        Array [
          Object {
            "text": "Ride 001",
            "value": "drum-samples/rides/ride_001.wav",
          },
          Object {
            "text": "Ride 002",
            "value": "drum-samples/rides/ride_002.wav",
          },
          Object {
            "text": "Ride 003",
            "value": "drum-samples/rides/ride_003.wav",
          },
          Object {
            "text": "Ride 004",
            "value": "drum-samples/rides/ride_004.wav",
          },
          Object {
            "text": "Ride 005",
            "value": "drum-samples/rides/ride_005.wav",
          },
          Object {
            "text": "Ride 006",
            "value": "drum-samples/rides/ride_006.wav",
          },
          Object {
            "text": "Ride 007",
            "value": "drum-samples/rides/ride_007.wav",
          },
        ]
      }
      drumType="ride"
      handleChange={[MockFunction]}
      instrumentId="4326432946537512"
    />
    <DrumModule
      drumData={
        Object {
          "pan": 0,
          "sample": "drum-samples/high-toms/high-tom_001.wav",
          "volume": 0,
        }
      }
      drumName="High Tom"
      drumSampleOptions={
        Array [
          Object {
            "text": "High Tom 001",
            "value": "drum-samples/high-toms/high-tom_001.wav",
          },
          Object {
            "text": "High Tom 002",
            "value": "drum-samples/high-toms/high-tom_002.wav",
          },
          Object {
            "text": "High Tom 003",
            "value": "drum-samples/high-toms/high-tom_003.wav",
          },
          Object {
            "text": "High Tom 004",
            "value": "drum-samples/high-toms/high-tom_004.wav",
          },
          Object {
            "text": "High Tom 005",
            "value": "drum-samples/high-toms/high-tom_005.wav",
          },
        ]
      }
      drumType="highTom"
      handleChange={[MockFunction]}
      instrumentId="4326432946537512"
    />
    <DrumModule
      drumData={
        Object {
          "pan": 0,
          "sample": "drum-samples/mid-toms/mid-tom_001.wav",
          "volume": 0,
        }
      }
      drumName="Mid Tom"
      drumSampleOptions={
        Array [
          Object {
            "text": "Mid Tom 001",
            "value": "drum-samples/mid-toms/mid-tom_001.wav",
          },
          Object {
            "text": "Mid Tom 002",
            "value": "drum-samples/mid-toms/mid-tom_002.wav",
          },
          Object {
            "text": "Mid Tom 003",
            "value": "drum-samples/mid-toms/mid-tom_003.wav",
          },
          Object {
            "text": "Mid Tom 004",
            "value": "drum-samples/mid-toms/mid-tom_004.wav",
          },
          Object {
            "text": "Mid Tom 005",
            "value": "drum-samples/mid-toms/mid-tom_005.wav",
          },
          Object {
            "text": "Mid Tom 006",
            "value": "drum-samples/mid-toms/mid-tom_006.wav",
          },
        ]
      }
      drumType="midTom"
      handleChange={[MockFunction]}
      instrumentId="4326432946537512"
    />
  </div>
  <div
    className="instrument-interface__module-container"
  >
    <DrumModule
      drumData={
        Object {
          "pan": 0,
          "sample": "drum-samples/low-toms/low-tom_001.wav",
          "volume": 0,
        }
      }
      drumName="Low Tom"
      drumSampleOptions={
        Array [
          Object {
            "text": "Low Tom 001",
            "value": "drum-samples/low-toms/low-tom_001.wav",
          },
          Object {
            "text": "Low Tom 002",
            "value": "drum-samples/low-toms/low-tom_002.wav",
          },
          Object {
            "text": "Low Tom 003",
            "value": "drum-samples/low-toms/low-tom_003.wav",
          },
          Object {
            "text": "Low Tom 004",
            "value": "drum-samples/low-toms/low-tom_004.wav",
          },
          Object {
            "text": "Low Tom 005",
            "value": "drum-samples/low-toms/low-tom_005.wav",
          },
        ]
      }
      drumType="lowTom"
      handleChange={[MockFunction]}
      instrumentId="4326432946537512"
    />
    <DrumModule
      drumData={
        Object {
          "pan": 0,
          "sample": "drum-samples/fx/fx_001.wav",
          "volume": 0,
        }
      }
      drumName="fx"
      drumSampleOptions={
        Array [
          Object {
            "text": "FX 001",
            "value": "drum-samples/fx/fx_001.wav",
          },
          Object {
            "text": "FX 002",
            "value": "drum-samples/fx/fx_002.wav",
          },
          Object {
            "text": "FX 003",
            "value": "drum-samples/fx/fx_003.wav",
          },
          Object {
            "text": "FX 004",
            "value": "drum-samples/fx/fx_004.wav",
          },
          Object {
            "text": "FX 005",
            "value": "drum-samples/fx/fx_005.wav",
          },
          Object {
            "text": "FX 006",
            "value": "drum-samples/fx/fx_006.wav",
          },
          Object {
            "text": "FX 007",
            "value": "drum-samples/fx/fx_007.wav",
          },
          Object {
            "text": "FX 008",
            "value": "drum-samples/fx/fx_008.wav",
          },
        ]
      }
      drumType="fx"
      handleChange={[MockFunction]}
      instrumentId="4326432946537512"
    />
  </div>
</div>
`;
